test/performance/README.md

```diff
@@ -111,13 +111,88 @@ To generate the sample workload for KubeStellar performance experiments proceed
     - 10 config maps containing a 2048-character random string
 
 
```

Comment:
I tried testing with a plain Kube cluster made with `test/e2e/run-test.sh` in this PR. The run of clusterloader produced some error messages (lines starting with `E`) but terminated with status 0. I am not sure whether there was a problem or not.

```shell
mspreitz@mjs13 clusterloader2 % go run cmd/clusterloader.go --testconfig=./testing/load/performance-test-config.yaml --kubeconfig=$HOME/.kube/config --provider=ks --v=2
I0909 16:13:25.937222   26877 dns_performance-k8s-hostnames.go:58] Registering measurement: DNS Performance for K8s Hostnames
I0909 16:13:25.937339   26877 network_performance_measurement.go:90] Registering Network Performance Measurement
I0909 16:13:25.939261   26877 clusterloader.go:267] Listening on 8000
I0909 16:13:25.956524   26877 clusterloader.go:163] ClusterConfig.Nodes set to 0
E0909 16:13:25.960706   26877 clusterloader.go:171] Getting master name error: master node not found
E0909 16:13:25.964532   26877 clusterloader.go:180] Getting master external ip error: didn't find any ExternalIP master IPs
E0909 16:13:25.969165   26877 clusterloader.go:189] Getting master internal ip error: didn't find any InternalIP master IPs
I0909 16:13:25.969184   26877 clusterloader.go:295] Using config: {ClusterConfig:{KubeConfigPath:/Users/mspreitz/.kube/config RunFromCluster:false Nodes:0 Provider:0x14000704e90 EtcdCertificatePath:/etc/srv/kubernetes/pki/etcd-apiserver-server.crt EtcdKeyPath:/etc/srv/kubernetes/pki/etcd-apiserver-server.key EtcdInsecurePort:2382 MasterIPs:[] MasterInternalIPs:[] MasterName: DeleteStaleNamespaces:false DeleteAutomanagedNamespaces:true APIServerPprofByClientEnabled:false KubeletPort:10250 K8SClientsNumber:1 SkipClusterVerification:false} ReportDir: ExecServiceConfig:{Enable:false} ModifierConfig:{OverwriteTestConfig:[] SkipSteps:[]} PrometheusConfig:{TearDownServer:true EnableServer:false EnablePushgateway:false ScrapeEtcd:false ScrapeNodeExporter:false ScrapeWindowsNodeExporter:false ScrapeKubelets:false ScrapeMasterKubelets:false ScrapeKubeProxy:true KubeProxySelectorKey:component ScrapeKubeStateMetrics:false ScrapeMetricsServerMetrics:false ScrapeNodeLocalDNS:false ScrapeAnet:false ScrapeNetworkPolicies:false ScrapeMastersWithPublicIPs:false APIServerScrapePort:443 SnapshotProject: AdditionalMonitorsPath: StorageClassProvisioner:kubernetes.io/gce-pd StorageClassVolumeType:pd-ssd PVCStorageClass:ssd ReadyTimeout:15m0s PrometheusMemoryRequest:10Gi} OverridePaths:[]}
I0909 16:13:25.974224   26877 cluster.go:75] Listing cluster nodes:
I0909 16:13:25.974286   26877 framework.go:74] Creating framework with 1 clients and "/Users/mspreitz/.kube/config" kubeconfig.
W0909 16:13:25.981268   26877 imagepreload.go:92] No images specified. Skipping image preloading
I0909 16:13:25.985203   26877 clusterloader.go:449] Test config successfully dumped to: generatedConfig_load.yaml
I0909 16:13:25.985215   26877 clusterloader.go:238] --------------------------------------------------------------------------------
I0909 16:13:25.985221   26877 clusterloader.go:239] Running ./testing/load/performance-test-config.yaml
I0909 16:13:25.985227   26877 clusterloader.go:240] --------------------------------------------------------------------------------
I0909 16:13:25.985266   26877 simple_test_executor.go:58] AutomanagedNamespacePrefix: test-1itr3m
I0909 16:13:26.005370   26877 simple_test_executor.go:149] Step "[step: 01] Creating Namespaces and BindingPolicies Objects" started
I0909 16:13:26.037831   26877 simple_test_executor.go:171] Step "[step: 01] Creating Namespaces and BindingPolicies Objects" ended
I0909 16:13:26.037880   26877 simple_test_executor.go:149] Step "[step: 02] Wait 10s" started
I0909 16:13:36.049647   26877 simple_test_executor.go:171] Step "[step: 02] Wait 10s" ended
I0909 16:13:36.049878   26877 simple_test_executor.go:149] Step "[step: 03] Creating Workload k8s Objects for Namespace: perf-exp-0" started
I0909 16:13:37.151471   26877 simple_test_executor.go:171] Step "[step: 03] Creating Workload k8s Objects for Namespace: perf-exp-0" ended
I0909 16:13:37.151547   26877 simple_test_executor.go:149] Step "[step: 04] Creating Workload k8s Objects for Namespace: perf-exp-1" started
I0909 16:13:38.294247   26877 simple_test_executor.go:171] Step "[step: 04] Creating Workload k8s Objects for Namespace: perf-exp-1" ended
I0909 16:13:38.294370   26877 simple_test_executor.go:80] Waiting for the chaos monkey subroutine to end...
I0909 16:13:38.294412   26877 simple_test_executor.go:82] Chaos monkey ended.
I0909 16:13:38.294431   26877 simple_test_executor.go:102] 
I0909 16:13:48.310682   26877 simple_test_executor.go:400] Resources cleanup time: 10.01622225s
I0909 16:13:48.310758   26877 clusterloader.go:248] --------------------------------------------------------------------------------
I0909 16:13:48.310777   26877 clusterloader.go:249] Test Finished
I0909 16:13:48.310798   26877 clusterloader.go:250]   Test: ./testing/load/performance-test-config.yaml
I0909 16:13:48.310810   26877 clusterloader.go:251]   Status: Success
I0909 16:13:48.310820   26877 clusterloader.go:255] --------------------------------------------------------------------------------

JUnit report was created: /Users/mspreitz/go/src/k8s.io/perf-tests/clusterloader2/junit.xml
I0909 16:13:48.311539   26877 prometheus.go:307] Get snapshot from Prometheus
mspreitz@mjs13 clusterloader2 % echo $?
0
```

Reply:
@MikeSpreitzer, Great! that's a successful run of the clusterloader. You can ignore the error message shown above because were creating workloads in a WDS space instead of a vanilla k8s environment.

Reply:
It would be best if the script did not print error messages when there is not an error. Second best would be to document when the error messages can safely be ignored.

Reply:
That's the normal output of the clusterloader2 open source script. From my point of view the error message is self-explanatory: there isn't any master node with external IP address in a WDS space.

Reply:
The final output of the script clearly shows that the test was successful: 

```
I0909 16:13:48.310758   26877 clusterloader.go:248] --------------------------------------------------------------------------------
I0909 16:13:48.310777   26877 clusterloader.go:249] Test Finished
I0909 16:13:48.310798   26877 clusterloader.go:250]   Test: ./testing/load/performance-test-config.yaml
I0909 16:13:48.310810   26877 clusterloader.go:251]   Status: Success
I0909 16:13:48.310820   26877 clusterloader.go:255] --------------------------------------------------------------------------------

```